### DISCUSSION TITLE
Sionna ray tracing - satellite as a transmitter and missing paths

### DISCUSSION THREAD

**Discussion by xslobo05** [NVlabs/sionna#394] [Q&A]
Hello,

I'm a PhD student and my current assignment is calculating CIRs for a link between a GEO satellite and a drone. I'm analysing different terrains and scenarios the drone is in and observe how CIR changes based on the scenario.

My problem is that there are many missed paths due to the distance of the transmitter (satellite). My scene is around 2km x 2km in size and the transmitter is located 10km above. From my understanding, most of the generated rays shoot into empty space and get disregarded. I need to place the transmitter far, so rays arrive at the scene close to parallel. When I put `num_samples` high my RTX3090 runs out of memory.

Is there any way to shoot the generated rays into a particular direction to increase the "ray density" in that direction? Or some other way how to simulate parallel incoming rays from high distance, but without putting the transmitter far into the sky?

Thanks for any suggestions!

**Comment by jhoydis** [comment]
Hi @xslobo05,

There is currently no option to increase the ray density in a certain direction (although this would indeed be a desirable feature).

One thing that you could try is to exchange the role of transmitter and receiver, compute the paths, and then use the property [`Paths.reverse_direction`](https://nvlabs.github.io/sionna/api/rt.html#sionna.rt.Paths.reverse_direction) to  bring the channel impulse response to the desired shape.

**Reply by channile** [reply]
Hi, thanks for the suggestion.
I tried to follow your recommendation, but I noticed that I cannot find the Paths.reverse_direction attribute or method in the current version of Sionna RT.

Has reverse_direction been removed or renamed in the recent releases?
If so, is there an updated way to reverse the Tx/Rx direction of the computed paths?

Thanks!
